pkg/httputil/logging.go

```diff
@@ -71,18 +75,33 @@ func DebugLoggingMiddleware(requestIDHeaderName string, fields logging.Fields) f
 			writer.Header().Set(requestIDHeaderName, reqID)
 			next.ServeHTTP(writer, r) // handle the request
 
-			logging.FromContext(r.Context()).WithFields(logging.Fields{
+			loggingFields := logging.Fields{
 				"took":        time.Since(startTime),
 				"status_code": writer.StatusCode,
 				"sent_bytes":  writer.ResponseSize,
-			}).Debug("HTTP call ended")
+			}
+			endLogMsg := "HTTP call ended"
+
+			switch strings.ToLower(middlewareLogLevel) {
+			case "debug", "null", "none":
+				logging.FromContext(r.Context()).WithFields(loggingFields).Debug(endLogMsg)
+			case "info":
+				logging.FromContext(r.Context()).WithFields(loggingFields).Info(endLogMsg)
+			case "warn", "warning":
+				logging.FromContext(r.Context()).WithFields(loggingFields).Warning(endLogMsg)
+			case "error":
+				logging.FromContext(r.Context()).WithFields(loggingFields).Error(endLogMsg)
+			case "panic":
+				logging.FromContext(r.Context()).WithFields(loggingFields).Panic(endLogMsg)
+			}
 		})
 	}
 }
 
 func LoggingMiddleware(requestIDHeaderName string, fields logging.Fields, traceRequestHeaders bool) func(next http.Handler) http.Handler {
-	if logging.Level() == "trace" {
+	loggingMiddlewareLevel := viper.GetString(config.LoggingAuditLogLevel)
```

Comment:
Usually, we extract the configuration value from the config structure. The config structure also expose a method in this case it can also parse the string and return the logging level